generator/osm2type.cpp

```diff
@@ -642,16 +642,26 @@ void PreprocessElement(OsmElement * p, CalculateOriginFnT const & calcOrg)
 
   bool isSubway = false;
   bool isLightRail = false;
+  bool isPlatform = false;
+  bool isStopPosition = false;
   bool isBus = false;
   bool isTram = false;
+
   bool isCapital = false;
 
+  bool isMultipolygon = false;
+
+  /// @todo Rearrange processing code with accumulating tags for: PT, Surface, Cuisine, Artwork, Memorial, ...
+  /// Process in separate components (with unit-tests), like DetermineSurface.
   TagProcessor(p).ApplyRules({
       {"bridge", "yes", [&layer] { layer = "1"; }},
       {"tunnel", "yes", [&layer] { layer = "-1"; }},
+      {"covered", "yes", [&layer] { layer = "-1"; }},
```

Comment:
Probably this one https://github.com/organicmaps/organicmaps/issues/2564
Can't say for sure if it is really needed.

Reply:
Now I remember, we had an issue with a building and a highway. Most of the covered tags are for the highways under buildings.
You can check locally here for example: https://www.openstreetmap.org/#map=19/48.14226/11.55020
Before and after. Probably, many thing has changed since our styles refactoring, but anyway ..

Reply:
For this particular case adding `layer=-1` changes nothing in the main 3D buildings mode.
But hides the "covered" roads completely in 2D.

Which is not nice neither for comprehension nor for navigation, especially given that some footways are rendered over the building still (they don't have `covered=yes`).

IMHO in this case we should show covered roads as tunnels (but w/o a "Tunnel" title as it'll be misleading).

Reply:
Yes, hiding highways is not good. Same as I don't like hiding highways under the bridge.

Reply:
Removed